src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8000,33 +8014,59 @@ private CommonForEachStatementSyntax ParseForEachStatement()
             }
 
             var openParen = this.EatToken(SyntaxKind.OpenParenToken);
+            var variable = ParseExpressionOrDeclaration(ParseTypeMode.Normal, feature: MessageID.IDS_FeatureTuples, permitTupleDesignation: true);
+            var @in = this.EatToken(SyntaxKind.InKeyword, ErrorCode.ERR_InExpected);
+            if (!IsValidForeachVariable(variable))
+            {
+                @in = this.AddError(@in, ErrorCode.ERR_BadForeachDecl);
```

Comment:
I am confused. First you say: "I'd prefer to do improvements in a separate PR." Then you say: "I do not believe there is any work here worth doing." Why do you think placing errors at the right location isn't worth doing?